src/types/excelsheet.rs

```diff
@@ -149,28 +149,162 @@ impl SelectedColumns {
+
+        if col.is_empty() {
+            return Err(InvalidParameters(
+                "a column should have at least one character, got none".to_string(),
+            )
+            .into());
+        }
+
+        col.chars()
+            //  iterating over all chars reversed, to have a power based on their rank
+            .rev()
+            .enumerate()
+            //  Parses every char, checks its position and returns its numeric equivalent based on
+            //  its rank. For example, AB becomes 27 (26 + 1)
+            .map(|(idx, col_chr)| {
+                let pos_in_alphabet = Self::ALPHABET
+                    .iter()
+                    .position(|chr| chr == &col_chr)
+                    .ok_or_else(|| {
+                        FastExcelError::from(InvalidParameters(format!(
+                            "Char is not a valid column name: {col_chr}"
+                        )))
+                    })?;
+
+                Ok(match idx {
+                    // in case it's the last char, just return its position
+                    0 => pos_in_alphabet,
+                    // otherwise, 26^idx * (position + 1)
+                    // For example, CBA is 2081:
+                    // A -> 0
+                    // B -> 26 (53^1 * (1 + 1))
+                    // C -> 2028 (26^2 * (2 + 1))
+                    _ => 26usize.pow(idx as u32) * (pos_in_alphabet + 1),
+                })
+            })
+            // Sums all previously obtained ranks
+            .try_fold(0usize, |acc, elem_result| {
+                elem_result.map(|elem| acc + elem)
+            })
+    }
+
+    fn col_indices_for_letter_range(col_range: &str) -> FastExcelResult<Vec<usize>> {
+        use FastExcelErrorKind::InvalidParameters;
+
+        let col_elements = col_range.split(':').collect::<Vec<_>>();
+        if col_elements.len() == 2 {
+            let start = Self::col_idx_for_col_as_letter(col_elements[0])
+                .with_context(|| format!("invalid start element for range \"{col_range}\""))?;
+            let end = Self::col_idx_for_col_as_letter(col_elements[1])
+                .with_context(|| format!("invalid end element for range \"{col_range}\""))?;
+
+            match start.cmp(&end) {
+                cmp::Ordering::Less => Ok((start..end).collect()),
+                cmp::Ordering::Greater => Err(InvalidParameters(format!(
+                    "end of range is before start: \"{col_range}\""
+                ))
+                .into()),
+                cmp::Ordering::Equal => {
+                    Err(InvalidParameters(format!("empty range: \"{col_range}\"")).into())
+                }
+            }
+        } else {
+            Err(InvalidParameters(format!(
+                "expected range to contain exactly 2 elements, got {n_elements}: \"{col_range}\"",
+                n_elements = col_elements.len()
+            ))
+            .into())
+        }
+    }
+}
+
+// Does not support columns beyond 26 for now
```

Comment:
It seems like you did the code to support it 🤔

Reply:
woops forgot to remove the comment :sweat_smile:

Reply:
bb18d94